Testcase test_rbd_block_pvc_snapshot failed

Failed test case:
tests/manage/pv_services/pvc_snapshot/test_rbd_block_pvc_snapshot.py::TestRbdBlockPvcSnapshot::test_rbd_block_pvc_snapshot


```
tests.manage.pv_services.pvc_snapshot.test_rbd_block_pvc_snapshot.TestRbdBlockPvcSnapshot.test_rbd_block_pvc_snapshot
 Error Details
AssertionError: Data corruption found
```
```
 log.info("Verifying md5sum on new pods")
        for pod_obj in restore_pod_objs:
            log.info(f"Verifying md5sum on pod {pod_obj.name}")
>           verify_data_integrity(
                pod_obj=pod_obj,
                file_name=pod_obj.get_storage_path(storage_type="block"),
                original_md5sum=pod_obj.pvc.md5sum,
                block=True,
            )

tests/manage/pv_services/pvc_snapshot/test_rbd_block_pvc_snapshot.py:194: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

pod_obj = <ocs_ci.ocs.resources.pod.Pod object at 0x7f676d25fd60>
file_name = '/dev/rbdblock'
original_md5sum = '5077be927d1060eafd2859ebe9b55fce', block = True

    def verify_data_integrity(pod_obj, file_name, original_md5sum, block=False):
        """
        Verifies existence and md5sum of file created from first pod
    
        Args:
            pod_obj (Pod): The object of the pod
            file_name (str): The name of the file for which md5sum to be calculated
            original_md5sum (str): The original md5sum of the file
            block (bool): True if the volume mode of PVC used on pod is 'Block'.
                file_name will be the devicePath in this case.
    
        Returns:
            bool: True if the file exists and md5sum matches
    
        Raises:
            AssertionError: If file doesn't exist or md5sum mismatch
        """
        file_path = file_name if block else get_file_path(pod_obj, file_name)
        assert check_file_existence(pod_obj, file_path), f"File {file_name} doesn't exists"
        current_md5sum = cal_md5sum(pod_obj, file_name, block)
        logger.info(f"Original md5sum of file: {original_md5sum}")
        logger.info(f"Current md5sum of file: {current_md5sum}")
>       assert current_md5sum == original_md5sum, "Data corruption found"
E       AssertionError: Data corruption found

ocs_ci/ocs/resources/pod.py:1185: AssertionError
```
RUN ID: 1691416825


This is similar to the issue https://bugzilla.redhat.com/show_bug.cgi?id=2028757#c9 (which is originally ceated for due to clone test case failure).
Adding --direct=1 in the fio command will solve the racy behaviour of the test case.

--- a/tests/manage/pv_services/pvc_snapshot/test_rbd_block_pvc_snapshot.py
+++ b/tests/manage/pv_services/pvc_snapshot/test_rbd_block_pvc_snapshot.py
@@ -76,6 +76,7 @@ class TestRbdBlockPvcSnapshot(ManageTest):
                 size=f"{self.pvc_size - 1}G",
                 io_direction="write",
                 runtime=60,
+                direct=1,
             )
         log.info("IO started on all pods")
 
@@ -203,7 +204,7 @@ class TestRbdBlockPvcSnapshot(ManageTest):
         # Run IO on new pods
         log.info("Starting IO on new pods")
         for pod_obj in restore_pod_objs:
-            pod_obj.run_io(storage_type="block", size="500M", runtime=15)
+            pod_obj.run_io(storage_type="block", size="500M", runtime=15, direct=1)
 
         # Wait for IO completion on new pods
         log.info("Waiting for IO completion on new pods")

